### DISCUSSION TITLE
Need help with adding custom mode

### DISCUSSION THREAD

**Discussion by siduck** [davatorium/rofi#1454] [Q&A] [answered]
Hi there , im trying to add a custom mode  and the script is [this](https://github.com/TimTinkers/rofi-iwd-menu/blob/master/rofi-wifi-menu.sh) . The issue is that , it gives me this error "dont run rofi inside rofi". Ik running this script standalone will make it work but i want to make a custom mode for this! Ig this  [part](https://github.com/TimTinkers/rofi-iwd-menu/blob/master/rofi-wifi-menu.sh#L71)  opens a new rofi instance 

Error : 
 
![image](https://user-images.githubusercontent.com/59060246/132725669-8b618c58-660b-4afc-bc32-fefe4c103de8.png)

**Comment by siduck** [comment, marked as answer]
running the script standalone works  

![image](https://user-images.githubusercontent.com/59060246/132726467-6699e8ff-07f6-4a6a-bbd7-095bd2b487b8.png)

**Reply by DaveDavenport** [reply]
You cannot call rofi inside rofi, that won't work.  (thats why it generates a message).
Also if it could run, it would add another window, not a mode.

For a custom mode see the manpage:
https://github.com/davatorium/rofi/blob/next/doc/rofi-script.5.markdown

and the few examples https://github.com/davatorium/rofi/blob/next/Examples/test_script_mode.sh
